### PR TITLE
added output_image argument to plot_design_matrix

### PR DESCRIPTION
**Pull request by bthirion**
Addresses #217.

### CODE REVIEW

File: nistats/reporting.py [hunk 1]
```diff
@@ -310,7 +313,10 @@ def plot_design_matrix(design_matrix, rescale=True, ax=None):
     ax.set_xticklabels(names, rotation=60, ha='right')
 
     plt.tight_layout()
-
+    if output_file is not None:
+        display.savefig(output_file)
+        display.close()
```
Comment (KamalakerDadi) [inline nistats/reporting.py]: Why not simply plt.close()?

**Comment by KamalakerDadi** [comment]
Any plans to expose this in one of the examples ?

File: nistats/reporting.py [hunk 2]
```diff
@@ -292,14 +296,13 @@ def plot_design_matrix(design_matrix, rescale=True, ax=None):
...
     _set_mpl_backend
-    import matplotlib.pyplot as plt
 
     # normalize the values per column for better visualization
     _, X, names = check_design_matrix(design_matrix)
     if rescale:
         X = X / np.maximum(1.e-12, np.sqrt(np.sum(X ** 2, 0)))  # pylint: disable=no-member
     if ax is None:
-        plt.figure()
+        display = plt.figure()
```
Comment (KamalakerDadi) [inline nistats/reporting.py]: Optional: May be ```display``` is not needed

**Comment by bthirion** [comment]
I think I have addressed your comments.

**Comment by bthirion** [comment]
Will probably merge with in the next 24hours unless I hear otherwise.

**Comment by KamalakerDadi** [comment]
No comments on my side.

**Comment by bthirion** [comment]
Merging now.

**Pull request merged**